### PR TITLE
WIP: Windows Service Support

### PR DESCRIPTION
**Pull request by rjemanuele**

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -99,7 +119,7 @@ local function start(...)
 
   local options = {}
   options.configFile = argv.args.c or constants:get('DEFAULT_CONFIG_PATH')
-
+  p(options.configFile)
```
Comment (rphillips) [inline main.lua]: debug print here

File: winsvcwrap.lua [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+--]]
+
+local table = require('table')
+local winsvc = require('winsvc')
+local winsvcaux = require('winsvcaux')
+local uv = require('uv')
+local logging = require('logging')
+local jsonStringify = require('json').stringify
+
+local exports = {}
```
Comment (rphillips) [inline winsvcwrap.lua]: we don't need to declare exports anymore, or return them at the end of the file

File: winsvcwrap.lua [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+  logging.infof('Report Service Status, %s', jsonStringify(svcStatus))
+  -- Report the status of the service to the SCM.
+  winsvc.SetServiceStatus(svcStatusHandle, svcStatus)
+end
+
+
+exports.SvcInstall = function(svcName, longName, desc, params)
+  local svcPath, err = winsvcaux.GetModuleFileName()
+  if svcPath == nil then
```
Comment (rphillips) [inline winsvcwrap.lua]: ``` if err then ```

File: winsvcwrap.lua [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    return
+  end
+
+  if params and params.args then
+    svcPath = svcPath .. ' ' .. table.concat(params.args, ' ')
+  end
+
+  -- Get a handle to the SCM database
+  local schSCManager, err = winsvc.OpenSCManager(nil, nil, winsvc.SC_MANAGER_ALL_ACCESS)
+  if schSCManager == nil then
```
Comment (rphillips) [inline winsvcwrap.lua]: same here... ```if err then```

File: winsvcwrap.lua (hunk 3 above)
Comment (rjemanuele) [inline winsvcwrap.lua]: i'm on the fence on this.  as written currently, its really close to the prototype service shown by the microsoft examples....  they prefer to check the actual return and then the error.

**Comment by rphillips** [comment]
+10

**Pull request merged**